tests/devices/qutrit_mixed/test_qutrit_mixed_sampling.py

```diff
@@ -0,0 +1,688 @@
+        state = math.array(two_qutrit_pure_state, like=interface)
+        samples = sample_state(state, 10)
+        assert samples.shape == (10, 2)
+        assert samples.dtype == np.int64
+        assert_correct_sampled_two_qutrit_pure_state(samples)
+
+    @pytest.mark.jax
+    def test_prng_key_as_seed_uses_sample_state_jax(self, mocker, two_qutrit_state):
+        """Tests that sample_state calls _sample_state_jax if the seed is a JAX PRNG key"""
+        import jax
+
+        spy = mocker.spy(qml.devices.qutrit_mixed.sampling, "_sample_state_jax")
+        state = qml.math.array(two_qutrit_state, like="jax")
+
+        # prng_key specified, should call _sample_state_jax
+        _ = sample_state(state, 10, prng_key=jax.random.PRNGKey(15))
+        # prng_key defaults to None, should NOT call _sample_state_jax
+        _ = sample_state(state, 10, rng=15)
+
+        spy.assert_called_once()
+
+    @pytest.mark.jax
+    def test_sample_state_jax(self, two_qutrit_pure_state):
+        """Tests that the returned samples are as expected when explicitly calling _sample_state_jax."""
+        import jax
+
+        state = qml.math.array(two_qutrit_pure_state, like="jax")
+
+        samples = _sample_state_jax(state, 10, prng_key=jax.random.PRNGKey(84))
+
+        assert samples.shape == (10, 2)
+        assert samples.dtype == np.int64
+        assert_correct_sampled_two_qutrit_pure_state(samples)
+
+    @pytest.mark.jax
+    def test_prng_key_determines_sample_state_jax_results(self, two_qutrit_pure_state):
+        """Test that setting the seed as a JAX PRNG key determines the results for _sample_state_jax"""
+        import jax
+
+        state = qml.math.array(two_qutrit_pure_state, like="jax")
+
+        samples = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(12))
+        samples2 = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(12))
+        samples3 = _sample_state_jax(state, shots=10, prng_key=jax.random.PRNGKey(13))
+
+        assert np.all(samples == samples2)
+        assert not np.allclose(samples, samples3)
+
+    def test_sample_state_custom_rng(self, two_qutrit_state):
+        """Tests that a custom RNG can be used with sample_state."""
+        custom_rng = np.random.default_rng(12345)
+        samples = sample_state(two_qutrit_state, 4, rng=custom_rng)
+        expected = [[0, 2], [1, 0], [2, 1], [1, 2]]
+        assert qml.math.allequal(samples, expected)
+
+    @flaky
+    def test_entangled_qutrit_samples_always_match(self):
+        """Tests that entangled qutrits are always in the same state."""
+        num_samples = 10000
+
+        bell_state_vector = np.array([[1, 0, 0, 0, 1, 0, 0, 0, 1]])
+        bell_state = get_dm_of_state(bell_state_vector, 2, 3)
+
+        samples = sample_state(bell_state, num_samples)
+        assert samples.shape == (num_samples, 2)
+        assert not any(samples[:, 0] ^ samples[:, 1])  # all samples are entangled
```

Comment:
Nice use of XOR!

Reply:
Thanks, it was similar to a qubit test though.